contracts/impls/trustable/WitnetRequestBoardTrustableBase.sol

```diff
@@ -0,0 +1,501 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.7.0 <0.9.0;
+pragma experimental ABIEncoderV2;
+
+import "../WitnetRequestBoardUpgradableBase.sol";
+import "../../data/WitnetBoardDataACLs.sol";
+import "../../interfaces/IWitnetRequestBoardAdmin.sol";
+import "../../interfaces/IWitnetRequestBoardAdminACLs.sol";
+import "../../patterns/Payable.sol";
+
+/// @title Witnet Request Board "trustable" base implementation contract.
+/// @notice Contract to bridge requests to Witnet Decentralized Oracle Network.
+/// @dev This contract enables posting requests that Witnet bridges will insert into the Witnet network.
+/// The result of the requests will be posted back to this contract by the bridge nodes too.
+/// @author The Witnet Foundation
+abstract contract WitnetRequestBoardTrustableBase
+    is 
+        Payable,
+        IWitnetRequestBoardAdmin,
+        IWitnetRequestBoardAdminACLs,        
+        WitnetBoardDataACLs,
+        WitnetRequestBoardUpgradableBase        
+{
+    using Witnet for bytes;
+    using Witnet for Witnet.Result;
+
+    uint256 internal constant _ESTIMATED_REPORT_RESULT_GAS = 102496;
```

Comment:
According to the README, the gas cost is higher now:

```
|  Contract                            ·  Method          ·  Min        ·  Max        ·  Avg        ·  # calls     ·  usd (avg)  │
|  WitnetRequestBoardTrustableDefault  ·  reportResult    ·     118352  ·     120501  ·     118948  ·          18  ·          -  │
```

The `102496` constant comes from this commit: d29fd4284728dbdd6fa55ec074bc619f68e3ae55 and it is the "max", so the new value should be `120501`.

Reply:
Right. The slightly higher cost is mainly derived from the fact that `timestamp` and `epoch` are now stored for each newly reported result. Will then update the `_ESTIMATE_REPORT_RESULT_GAS` to `120501`. Should we not scare gas consumption, the `_reportResult` can be changed as to avoid writing these values to storage.

Reply:
Also, currently this constant is unique for all the different implementations. Maybe it should be defined somewhere else to allow supporting other EVM-compatible chains with different gas costs.